server/vulnerabilities/vuln_centos/centos.go

```diff
@@ -0,0 +1,512 @@
+// Package vuln_centos contains a ParseCentOSRepository method to parse the CentOS repository
+// to look out for CentOS releases that patch CVEs. It parses the changelogs from the metadata.
+//
+// It also contains a LoadCentOSFixedCVEs to load the results of the parsing.
+//
+// Both the parsing and loading of results use sqlite3 as backend storage.
+package vuln_centos
+
+import (
+	"compress/bzip2"
+	"context"
+	"database/sql"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gocolly/colly"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/ulikunitz/xz"
+)
+
+// CentOSPkg holds data to identify a CentOS package.
+type CentOSPkg struct {
+	Name    string
+	Version string
+	Release string
+	Arch    string
+}
+
+// String implements fmt.Stringer.
+func (p CentOSPkg) String() string {
+	return p.Name + "-" + p.Version + "-" + p.Release + "." + p.Arch
+}
+
+// FixedCVESet is a set of fixed CVEs.
+type FixedCVESet map[string]struct{}
+
+// CentOSPkgSet is a set of CentOS packages and their fixed CVEs.
+type CentOSPkgSet map[CentOSPkg]FixedCVESet
+
+// Add adds the given package and CVE/s to the set.
+func (p CentOSPkgSet) Add(pkg CentOSPkg, fixedCVEs ...string) {
+	s := p[pkg]
+	if s == nil {
+		s = make(FixedCVESet)
+	}
+	for _, fixedCVE := range fixedCVEs {
+		s[fixedCVE] = struct{}{}
+	}
+	p[pkg] = s
+}
+
+const centOSPkgsCVEsTable = "centos_pkgs_fixed_cves"
```

Comment:
Why is this a const instead of just a string in the queries?

Reply:
Fixed. It now errors out if the table is not there.